File: Iteration-2/src/Transcript.java

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

public class Transcript {
    private String studentID;
    private double GPA;
    private List<String> completedCourses;
    private List<String> failedCourses;
    private int completedCredits;
    private int creditsTaken;

    public Transcript(String studentID) {
        this.studentID = studentID;
        this.GPA = 0.0;
        this.completedCourses = new ArrayList<>();
        this.failedCourses = new ArrayList<>();
        this.completedCredits = 0;
        this.creditsTaken = 0;
    }

	public String getStudentID() {
		return studentID;
	}
	
	public void setStudentID(String studentID) {
		this.studentID = studentID;
	}
	
	public double getGPA() {
		return GPA;
	}
	
	public void setGPA(double gPA) {
		GPA = gPA;
	}
	
	public List<String> getCompletedCourses() {
		return completedCourses;
	}

	public void setCompletedCourses(List<String> list) {
		this.completedCourses = list;
	}

	public List<String> getFailedCourses() {
		return failedCourses;
	}

	public void setFailedCourses(List<String> list) {
		this.failedCourses = list;
	}
	
	public int getCompletedCredits() {
		return completedCredits;
	}
	
	public void setCompletedCredits(int completedCredits) {
		this.completedCredits = completedCredits;
	}

	public int getCreditsTaken() {
		return creditsTaken;
	}
	
	public void setCreditsTaken(int creditsTaken) {
		this.creditsTaken = creditsTaken;
	}

	public void addCompletedCourse(Course course) {
        completedCourses.addAll((Collection<? extends String>) course);
        completedCredits += course.getCredit();
        creditsTaken += course.getCredit();
    }

    public void addFailedCourse(Course course) {
        failedCourses.addAll((Collection<? extends String>) course);
        creditsTaken += course.getCredit();  
    }

}

File: Iteration-3/Src/Transcript.py
class Transcript:
    def __init__(self, student_id):
        self.student_id = student_id
        self.gpa = 0.0
        self.completed_courses = []
        self.failed_courses = []
        self.completed_credits = 0
        self.credits_taken = 0

    def get_student_id(self):
        return self.student_id

    def set_student_id(self, student_id):
        self.student_id = student_id

    def get_gpa(self):
        return self.gpa

    def set_gpa(self, gpa):
        self.gpa = gpa

    def get_completed_courses(self):
        return self.completed_courses

    def set_completed_courses(self, course_list):
        self.completed_courses = course_list

    def get_failed_courses(self):
        return self.failed_courses

    def set_failed_courses(self, course_list):
        self.failed_courses = course_list

    def get_completed_credits(self):
        return self.completed_credits

    def set_completed_credits(self, completed_credits):
        self.completed_credits = completed_credits

    def get_credits_taken(self):
        return self.credits_taken

    def set_credits_taken(self, credits_taken):
        self.credits_taken = credits_taken

    def add_completed_course(self, course):
        self.completed_courses.append(course)
        self.completed_credits += course.get_credit()
        self.credits_taken += course.get_credit()

    def add_failed_course(self, course):
        self.failed_courses.append(course)
        self.credits_taken += course.get_credit()
